packages/react-admin/src/sideEffect/saga/auth.js

```diff
@@ -33,7 +36,8 @@ export default authClient => {
                         type: USER_LOGIN_SUCCESS,
                         payload: authPayload,
                     });
-                    yield put(push(meta.pathName || '/'));
+                    const redirectTo = yield select(nextPathnameSelector);
```

Comment:
why that change? You're breaking a feature that was added to redirect to a specific page after login (`meta.pathName`).

Reply:
Also, I'm afraid than when the `nextPathnameSelector` function is called, the routing state is already in `/login`, so it doesn't make sense to redirect there?

Reply:
> why that change? You're breaking a feature that was added to redirect to a specific page after login (meta.pathName).

I wanted to remove the dependency of https://github.com/marmelab/admin-on-rest/blob/d82fb2dc9df9d75c5922197a07fd49dfe977128d/packages/react-admin/src/mui/auth/Login.js#L102 in the `loginForm` and I decided that it was best to put the logic to resolve the previous location in the saga instead of the component. The previous location state is reduced so available to select. But ignoring `meta.pathName` is an oversight, fixed it. 

> Also, I'm afraid than when the nextPathnameSelector function is called, the routing state is already in /login, so it doesn't make sense to redirect there?

Yes, but the previous location state is stored separately, @see: https://github.com/marmelab/admin-on-rest/blob/d82fb2dc9df9d75c5922197a07fd49dfe977128d/packages/react-admin/src/sideEffect/saga/auth.js#L59 so it is up for grabs.

Reply:
Then you need to change the `userLogin` action signature, too.